### PR TITLE
Filter away low confidence lines

### PR DESCRIPTION
**Pull request by stinebuu**

### CODE REVIEW

File: docs/workflows/transitions/preprocess/preprocess/utils.py [hunk 1]
```diff
@@ -211,3 +217,24 @@ def patch_empty_predictions(predictions, labels, no_empty_prediction_fields):
...
+            for index, line in enumerate(prediction['value']):
+                line = filter_by_top1(line, column_names[label])
+                if not line:
+                    continue
+                # column names that are not present in the line counts as 0% confidence
+                line_columns_present = {line_dict['label'] for line_dict in line}
+                line += [{'confidence': 0.0} for _ in column_names[label] - line_columns_present]
+                average_confidence = np.mean([line_dict['confidence'] for line_dict in line])
+                if average_confidence < MINIMUM_AVERAGE_LINE_CONFIDENCE:
+                    prediction['value'][index] = []
```
Comment (magnurud) [inline docs/workflows/transitions/preprocess/preprocess/utils.py:238]: Will this be returned as an empty line, or will it be pruned away before it is returned? 
We want it to be pruned away.

**Review by magnurud — APPROVED** [review]

File: docs/workflows/transitions/preprocess/preprocess/utils.py [hunk 2]
```diff
@@ -211,3 +217,27 @@ def patch_empty_predictions(predictions, labels, no_empty_prediction_fields):
...
+    }
+
+    for prediction in predictions:
+        label = prediction['label']
+        if field_config.get(label, {}).get('type') == 'lines':
+            line_predictions = []
+            for line in prediction['value']:
+                top_1_line = filter_by_top1(line, column_names[label])
+                if not top_1_line:
+                    continue
```
Comment (tringholm) [inline docs/workflows/transitions/preprocess/preprocess/utils.py:233]: This should not be necessary

File: docs/workflows/transitions/preprocess/preprocess/utils.py [hunk 3]
```diff
@@ -211,3 +217,27 @@ def patch_empty_predictions(predictions, labels, no_empty_prediction_fields):
...
+                # column names that are not present in the line counts as 0% confidence
+                line_columns_present = {line_dict['label'] for line_dict in top_1_line}
```
Comment (tringholm) [inline docs/workflows/transitions/preprocess/preprocess/utils.py:235]: A suggestion for decreasing verbosity - and also, I don't think `line_dict` is a good name for a prediction - I would prefer `prediction` if not `p`.
```suggestion
                line_columns_present = {p['label'] for p in top_1_line}
```

File: docs/workflows/transitions/preprocess/preprocess/utils.py [hunk 4]
```diff
@@ -211,3 +217,27 @@ def patch_empty_predictions(predictions, labels, no_empty_prediction_fields):
...
+        label: {column_name for column_name in config['fields']}
+        for label, config in field_config.items() if config.get('type') == 'lines'
+    }
+
+    for prediction in predictions:
+        label = prediction['label']
+        if field_config.get(label, {}).get('type') == 'lines':
+            line_predictions = []
+            for line in prediction['value']:
+                top_1_line = filter_by_top1(line, column_names[label])
```
Comment (tringholm) [inline docs/workflows/transitions/preprocess/preprocess/utils.py:231]: If I understand correctly how this works, I think this is a clearer name.
```suggestion
                top_1_predictions = filter_by_top1(line, column_names[label])
```

File: docs/workflows/transitions/preprocess/preprocess/utils.py [hunk 5]
```diff
@@ -228,13 +228,12 @@ def filter_away_low_confidence_lines(predictions, field_config):
         if field_config.get(label, {}).get('type') == 'lines':
             line_predictions = []
             for line in prediction['value']:
-                top_1_line = filter_by_top1(line, column_names[label])
-                if not top_1_line:
-                    continue
+                top_1_prediction = filter_by_top1(line, column_names[label])
```
Comment (tringholm) [inline docs/workflows/transitions/preprocess/preprocess/utils.py:231]:
```suggestion
                top_1_predictions = filter_by_top1(line, column_names[label])
```

**Review by tringholm — APPROVED** [review]

**Pull request merged**